exporter/views.py

```diff
@@ -120,9 +129,22 @@ def get(self, request, *args, **kwargs):
         # Download zip
         with open("%s.zip" % export_base_dir, "rb") as zip_file:
             response = HttpResponse(zip_file, content_type="application/zip")
-        response["Content-Disposition"] = "attachment; filename=%s.zip" % campaign_slug
+
+        export_filename = "%s-%s.zip" % (
+            campaign_slug,
+            datetime.today().isoformat(timespec="minutes"),
+        )
+
+        response["Content-Disposition"] = "attachment; filename=%s" % export_filename
 
         # Upload zip to S3 bucket
+        s3_bucket = getattr(settings, "S3_BUCKET_NAME", None)
+
+        if s3_bucket:
+            s3 = boto3.resource("s3")
+            s3.Bucket(s3_bucket).upload_file(
```

Comment:
As far as I can tell, we're uploading the file but not using it. That raises the possibility of things like not setting the MIME type or security options on the file without noticing it. I'm wondering whether the process should basically be:

```python
try:
    if s3_bucket:
        # do upload
        # return redirect to the S3 object's public URL in the response returned by `upload_file`
    else:
    return response
finally:
    shutil.rmtree(export_base_dir)
```

Reply:
`shutil.rmtree` isn't necessary with `TemporaryDirectory` so it's even easier than this.